src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1556,9 +1556,12 @@ else if (logger.isDebugEnabled())
                         RowDataResolver resolver = (RowDataResolver)handler.resolver;
                         try
                         {
-                            // wait for the repair writes to be acknowledged, to minimize impact on any replica that's
-                            // behind on writes in case the out-of-sync row is read multiple times in quick succession
-                            FBUtilities.waitOnFutures(resolver.repairResults, DatabaseDescriptor.getWriteRpcTimeout());
+                            if (!DatabaseDescriptor.getDisableBlockOnReadRepair())
+                            {
+                                // wait for the repair writes to be acknowledged, to minimize impact on any replica that's
+                                // behind on writes in case the out-of-sync row is read multiple times in quick succession
+                                FBUtilities.waitOnFutures(resolver.repairResults, DatabaseDescriptor.getWriteRpcTimeout());
+                            }
```

Comment:
If that flag is set, do we need to disable waiting on repair results in `getRangeSlice` as well? 
https://github.com/palantir/cassandra/blob/d21f3feefde21b5b35535a2639f1a408eab947bb/src/java/org/apache/cassandra/service/StorageProxy.java#L1970

Reply:
You can trace this through `RangeSliceResponseResolver`